diff --git a/drivers/mmc/sdhci.c b/drivers/mmc/sdhci.c
index 7845f87..efa011e 100644
--- a/drivers/mmc/sdhci.c
+++ b/drivers/mmc/sdhci.c
@@ -479,7 +479,7 @@ int add_sdhci(struct sdhci_host *host, u32 max_clk, u32 min_clk)
 	if (host->quirks & SDHCI_QUIRK_BROKEN_VOLTAGE)
 		mmc->voltages |= host->voltages;
 
-	mmc->host_caps = MMC_MODE_HS | MMC_MODE_HS_52MHz | MMC_MODE_4BIT;
+	mmc->host_caps = MMC_MODE_4BIT;
 	if (caps & SDHCI_CAN_DO_8BIT)
 		mmc->host_caps |= MMC_MODE_8BIT;
 	if (host->host_caps)
