diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index d0fbc88..de27c88 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -22,6 +22,10 @@
 #undef CONFIG_ZLIB
 #undef CONFIG_SYS_HUSH_PARSER
 #undef CONFIG_CMD_NET
+ 
+#define CONFIG_CMD_ELF
+#define CONFIG_API
+#define CONFIG_SYS_MMC_MAX_DEVICE 1
 
 #include <asm/arch/cpu.h>
 #include <asm/arch/hardware.h>
