diff --git a/api/api_storage.c b/api/api_storage.c
index c535712..fcc686d 100644
--- a/api/api_storage.c
+++ b/api/api_storage.c
@@ -124,10 +124,13 @@ static int dev_stor_get(int type, int first, int *more, struct device_info *di)
 
 	if (first) {
 		di->cookie = (void *)get_dev(specs[type].name, 0);
-		if (di->cookie == NULL)
+		if (di->cookie == NULL) {
 			return 0;
-		else
+		} else {
 			found = 1;
+			if (specs[type].max_dev > 1)
+				*more = 1;
+		}
 
 	} else {
 		for (i = 0; i < specs[type].max_dev; i++)
@@ -163,7 +166,8 @@ static int dev_stor_get(int type, int first, int *more, struct device_info *di)
 			dd = (block_dev_desc_t *)di->cookie;
 			if (dd->type == DEV_TYPE_UNKNOWN) {
 				debugf("device instance exists, but is not active..");
-				found = 0;
+				di->di_stor.block_count = 0;
+				di->di_stor.block_size = 0;
 			} else {
 				di->di_stor.block_count = dd->lba;
 				di->di_stor.block_size = dd->blksz;
