diff --git a/api/api.c b/api/api.c
index a3bf60a..58263cf 100644
--- a/api/api.c
+++ b/api/api.c
@@ -226,8 +226,8 @@ static int API_dev_enum(va_list ap)
 		debugf("RESTART ENUM\n");
 
 		/* net device enumeration first */
-		if (dev_enum_net(di))
-			return 0;
+		//if (dev_enum_net(di))
+		//return 0;
 	}
 
 	/*
@@ -264,8 +264,8 @@ static int API_dev_open(va_list ap)
 	if (di->type & DEV_TYP_STOR)
 		err = dev_open_stor(di->cookie);
 
-	else if (di->type & DEV_TYP_NET)
-		err = dev_open_net(di->cookie);
+	//	else if (di->type & DEV_TYP_NET)
+	//		err = dev_open_net(di->cookie);
 	else
 		err = API_ENODEV;
 
@@ -295,8 +295,8 @@ static int API_dev_close(va_list ap)
 	if (di->type & DEV_TYP_STOR)
 		err = dev_close_stor(di->cookie);
 
-	else if (di->type & DEV_TYP_NET)
-		err = dev_close_net(di->cookie);
+	//	else if (di->type & DEV_TYP_NET)
+	//		err = dev_close_net(di->cookie);
 	else
 		/*
 		 * In case of unknown device we cannot change its state, so
@@ -364,8 +364,8 @@ static int API_dev_write(va_list ap)
 		 */
 		return API_ENODEV;
 
-	else if (di->type & DEV_TYP_NET)
-		err = dev_write_net(di->cookie, buf, *len);
+	//	else if (di->type & DEV_TYP_NET)
+	//		err = dev_write_net(di->cookie, buf, *len);
 	else
 		err = API_ENODEV;
 
@@ -436,6 +436,7 @@ static int API_dev_read(va_list ap)
 
 		*act_len_stor = dev_read_stor(di->cookie, buf, *len_stor, *start);
 
+#if 0
 	} else if (di->type & DEV_TYP_NET) {
 
 		/* 3. arg points to the var with length of packet to read */
@@ -452,6 +453,7 @@ static int API_dev_read(va_list ap)
 
 		*act_len_net = dev_read_net(di->cookie, buf, *len_net);
 
+#endif
 	} else
 		return API_ENODEV;
 
