--- a/include/configs/ti_am335x_common.h	2014-04-10 09:22:45.000000000 +0800
+++ b/include/configs/ti_am335x_common.h	2014-04-10 09:23:01.000000000 +0800
@@ -48,10 +48,6 @@
  */
 #define CONFIG_SYS_BOOTCOUNT_ADDR	0x44E3E000
 
-/* Enable the HW watchdog, since we can use this with bootcount */
-#define CONFIG_HW_WATCHDOG
-#define CONFIG_OMAP_WATCHDOG
-
 /*
  * SPL related defines.  The Public RAM memory map the ROM defines the
  * area between 0x402F0400 and 0x4030B800 as a download area and
@@ -62,9 +58,6 @@
 #define CONFIG_SPL_TEXT_BASE		0x402F0400
 #define CONFIG_SPL_MAX_SIZE		(0x4030B800 - CONFIG_SPL_TEXT_BASE)
 
-/* Enable the watchdog inside of SPL */
-#define CONFIG_SPL_WATCHDOG_SUPPORT
-
 /*
  * Since SPL did pll and ddr initialization for us,
  * we don't need to do it twice.
