diff --git a/arch/arm/cpu/armv7/omap-common/lowlevel_init.S b/arch/arm/cpu/armv7/omap-common/lowlevel_init.S
index 90b3c8a..9961dbe 100644
--- a/arch/arm/cpu/armv7/omap-common/lowlevel_init.S
+++ b/arch/arm/cpu/armv7/omap-common/lowlevel_init.S
@@ -32,6 +32,7 @@
 #include <linux/linkage.h>
 
 ENTRY(save_boot_params)
+#ifdef CONFIG_SPL_BUILD
 	/*
 	 * See if the rom code passed pointer is valid:
 	 * It is not valid if it is not in non-secure SRAM
@@ -44,7 +45,13 @@ ENTRY(save_boot_params)
 	ldr	r2, =NON_SECURE_SRAM_END
 	cmp	r2, r0
 	blt	1f
-
+#else
+	/*
+	 * SPL passes U-boot the address of the boot param pointer
+	 */
+	ldr	r0, [r0]
+#endif
+	
 	/*
 	 * store the boot params passed from rom code or saved
 	 * and passed by SPL
@@ -53,13 +60,13 @@ ENTRY(save_boot_params)
 	beq	1f
 	ldr	r1, =boot_params
 	str	r0, [r1]
-#ifdef CONFIG_SPL_BUILD
+
 	/* Store the boot device in spl_boot_device */
 	ldrb	r2, [r0, #BOOT_DEVICE_OFFSET]	@ r1 <- value of boot device
 	and	r2, #BOOT_DEVICE_MASK
 	ldr	r3, =boot_params
 	strb	r2, [r3, #BOOT_DEVICE_OFFSET]	@ spl_boot_device <- r1
-
+#ifdef CONFIG_SPL_BUILD
 	/*
 	 * boot mode is only valid for device that can be raw or FAT booted.
 	 * in other cases it may be fatal to look.  While platforms differ
