diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index d0fbc88..de27c88 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -41,9 +45,14 @@
 #define CONFIG_SYS_AUTOLOAD		"no"
 #define CONFIG_BOOTFILE			"uImage"
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"verify=yes\0" \
 	"ramdisk_file=ramdisk.gz\0" \
 
+#define CONFIG_BOOTCOMMAND \
+	"mmc rescan;" \
+	"fatload mmc 0:1 88000000 ubldr;" \
+	"bootelf 88000000"
+
 /* Clock Defines */
 #define V_OSCK				24000000  /* Clock output from T2 */
 #define V_SCLK				(V_OSCK)
