--- ../u-boot-2014.07/include/configs/wandboard.h	2014-07-14 11:16:45.000000000 -0600
+++ u-boot-2014.07/include/configs/wandboard.h	2014-08-29 20:59:42.312926255 -0600
@@ -18,6 +18,15 @@
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
 
+#define CONFIG_SYS_ICACHE_OFF                                                                             
+#define CONFIG_SYS_DCACHE_OFF                                                                             
+                                                                                                          
+#ifndef CONFIG_SPL_BUILD                                                                                  
+#define CONFIG_CMD_ELF                                                                                    
+#define CONFIG_SYS_MMC_MAX_DEVICE 4
+#define CONFIG_API                                                                                        
+#endif  
+
 #define MACH_TYPE_WANDBOARD		4412
 #define CONFIG_MACH_TYPE		MACH_TYPE_WANDBOARD
 
