diff --git a/fs/fat/fat.c b/fs/fat/fat.c
index 0206b14..4c7dfec 100644
--- a/fs/fat/fat.c
+++ b/fs/fat/fat.c
@@ -319,11 +319,11 @@ get_cluster(fsdata *mydata, __u32 clustnum, __u8 *buffer, unsigned long size)
  * into 'buffer'.
  * Return the number of bytes read or -1 on fatal errors.
  */
 #ifndef CONFIG_ZYNQ_OCM
 #if defined(CONFIG_ZYNQ) && defined(CONFIG_SPL_BUILD)
-__section(.ddr)
+__section(".ddr")
 #endif
 __u8 get_contents_vfatname_block[MAX_CLUSTSIZE]
 	__aligned(ARCH_DMA_MINALIGN);
 #else
 __u8 *get_contents_vfatname_block = (__u8 *)FAT_BUFF_PTR_OCM;
@@ -578,11 +578,11 @@ static __u8 mkcksum(const char name[8], const char ext[3])
  * Get the directory entry associated with 'filename' from the directory
  * starting at 'startsect'
  */
 #ifndef CONFIG_ZYNQ_OCM
 #if defined(CONFIG_ZYNQ) && defined(CONFIG_SPL_BUILD)
-__section(.ddr)
+__section(".ddr")
 #endif
 __u8 get_dentfromdir_block[MAX_CLUSTSIZE]
 	__aligned(ARCH_DMA_MINALIGN);
 #endif
 
@@ -819,11 +819,11 @@ exit:
 	return ret;
 }
 
 #ifndef CONFIG_ZYNQ_OCM
 #if defined(CONFIG_ZYNQ) && defined(CONFIG_SPL_BUILD)
-__section(.ddr)
+__section(".ddr")
 #endif
 __u8 do_fat_read_at_block[MAX_CLUSTSIZE]
 	__aligned(ARCH_DMA_MINALIGN);
 #endif
 
