diff --git a/arch/arm/cpu/armv7/zynq/config.mk b/arch/arm/cpu/armv7/zynq/config.mk
index 9640433..f53a9f2 100644
--- a/arch/arm/cpu/armv7/zynq/config.mk
+++ b/arch/arm/cpu/armv7/zynq/config.mk
@@ -2,9 +2,5 @@
 # Copyright (C) 2013 Xilinx, Inc. All rights reserved.
 #
 # SPDX-License-Identifier:	GPL-2.0+
 #
 PLATFORM_RELFLAGS += -fno-strict-aliasing
-# Xilinx, added to prevent unaligned accesses which started happening # with GCC 4.5.2 tools
-PLATFORM_RELFLAGS += -mno-unaligned-access
-# Allow NEON instructions (needed for lowlevel_init.S with GNU toolchain)
-PLATFORM_RELFLAGS += -mfpu=neon
diff --git a/arch/arm/cpu/armv7/zynq/lowlevel_init.S b/arch/arm/cpu/armv7/zynq/lowlevel_init.S
index 6d714b7..a6494d3 100644
--- a/arch/arm/cpu/armv7/zynq/lowlevel_init.S
+++ b/arch/arm/cpu/armv7/zynq/lowlevel_init.S
@@ -13,14 +13,15 @@ ENTRY(lowlevel_init)
 	/* Enable the the VFP */
 	mrc	p15, 0, r1, c1, c0, 2
 	orr	r1, r1, #(0x3 << 20)
 	orr	r1, r1, #(0x3 << 20)
 	mcr	p15, 0, r1, c1, c0, 2
-	isb
+/*	isb
 	fmrx	r1, FPEXC
 	orr	r1,r1, #(1<<30)
 	fmxr	FPEXC, r1
+*/
 
 	/* Move back to caller */
 	mov	pc, lr
 
 ENDPROC(lowlevel_init)
