diff --git a/examples/api/Makefile b/examples/api/Makefile
index 6cf23d1..1d1f18d 100644
--- a/examples/api/Makefile
+++ b/examples/api/Makefile
@@ -35,11 +35,11 @@ targets += $(OBJS)
 OBJS := $(addprefix $(obj)/,$(OBJS))
 
 #########################################################################
 
 quiet_cmd_link_demo = LD      $@
-cmd_link_demo = $(LD) --gc-sections -Ttext $(LOAD_ADDR) -o $@ $(filter-out $(PHONY), $^) $(PLATFORM_LIBS)
+cmd_link_demo = $(LD) --gc-sections -static -Ttext $(LOAD_ADDR) -o $@ $(filter-out $(PHONY), $^) $(PLATFORM_LIBS)
 
 $(obj)/demo: $(OBJS) FORCE
 	$(call if_changed,link_demo)
 
 # demo.bin is never genrated. Is this necessary?
