diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index e0ac494..fa82bb7 100755
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -13,6 +13,10 @@
 #ifndef __CONFIG_AM335X_EVM_H
 #define __CONFIG_AM335X_EVM_H
 
+#define CONFIG_CMD_ELF
+#define CONFIG_API
+#define CONFIG_SYS_MMC_MAX_DEVICE	1
+
 #define CONFIG_AM335X
 #define CONFIG_TI81XX
 #define CONFIG_SYS_NO_FLASH
